src/kernel/mreader.ml

```diff
@@ -14,6 +14,7 @@ type result = {
   comments      : comment list;
   parsetree     : parsetree;
   no_labels_for_completion : bool;
+  keywords      : string list;
```

Comment:
I don't think this needs to be stored in there. That list is going to stay the same across all the parse / buffers / etc.